src/utils/eventHandlers/setupGlobalEventHandlers.js

```diff
@@ -0,0 +1,249 @@
+
+  const state = productListStore.getState();
+  select.value = String(state.limit);
+
+  select.addEventListener("change", async (e) => {
+    const newLimit = parseInt(e.target.value, 10);
+    if (!isNaN(newLimit) && newLimit !== state.limit) {
+      productListStore.setLimit(newLimit);
+      updateQueryParams({ limit: newLimit }, true);
+      await renderInitialContent();
+    }
+  });
+};
+
+/**
+ * 무한 스크롤 이벤트 리스너 설정
+ */
+const setupInfiniteScroll = (loadMoreProducts) => {
+  let isThrottled = false;
+  const throttleDelay = 100;
+
+  const handleScroll = () => {
+    // 상품 목록 페이지에서만 무한 스크롤 활성화
+    const isProductListPage = window.location.pathname === "/" || window.location.pathname === "";
+    if (!isProductListPage) return;
+
+    if (isThrottled) return;
+
+    isThrottled = true;
+    setTimeout(() => {
+      isThrottled = false;
+    }, throttleDelay);
+
+    const state = productListStore.getState();
+
+    if (state.isLoading || !state.hasMore) return;
+
+    const scrollTop = window.pageYOffset || document.documentElement.scrollTop;
+    const windowHeight = window.innerHeight;
+    const documentHeight = document.documentElement.scrollHeight;
+
+    // 스크롤이 페이지 하단 80%에 도달했을 때 다음 페이지 로드
+    if (scrollTop + windowHeight >= documentHeight * 0.8) {
+      loadMoreProducts();
+    }
+  };
+
+  window.addEventListener("scroll", handleScroll);
+};
+
+/**
+ * 상품 카드 이벤트 리스너 설정
+ */
+const setupProductCardEventListeners = () => {
+  // 썸네일(이미지) 클릭 이벤트 (상세 페이지로 이동)
+  document.addEventListener("click", (e) => {
+    const productImage = e.target.closest(".product-image");
+    if (productImage) {
+      e.stopPropagation(); // 이벤트 버블링 방지
+      const productId = productImage.dataset.productId;
+      if (productId) {
+        navigateToProduct(productId);
+      }
+    }
+  });
+
+  // 장바구니 버튼 클릭 이벤트
+  document.addEventListener("click", (e) => {
+    const addToCartBtn = e.target.closest(".add-to-cart-btn");
+    if (addToCartBtn) {
+      e.stopPropagation(); // 이벤트 버블링 방지
+
+      const productCard = addToCartBtn.closest(".product-card");
+      if (productCard) {
+        const productData = JSON.parse(productCard.dataset.product);
+        if (productData) {
+          // 장바구니에 추가 (cartHandler의 addToCart 함수 사용)
+          import("../cartHandler.js").then(({ addToCart }) => {
+            addToCart(productData);
+          });
+        }
+      }
+    }
+  });
+};
+
+// 검색 이벤트 리스너 설정 함수
+export const setupSearchEventListeners = (renderCallback) => {
+  const searchInput = document.getElementById("search-input");
+  const searchButton = document.getElementById("search-button");
+  const clearButton = document.getElementById("clear-search-button");
+
+  if (!searchInput) return;
+
+  // 검색 버튼 클릭 이벤트
+  if (searchButton) {
+    searchButton.addEventListener("click", () => {
+      const searchTerm = searchInput.value.trim();
+      if (searchTerm) {
+        performSearch(searchTerm, renderCallback);
+      }
+    });
+  }
```

Comment:
소소하게,,, ? 를 쓰면 if 문을 생략할 수 있습니다.

```suggestion
  searchButton?.addEventListener("click", () => {
    const searchTerm = searchInput.value.trim();
    if (searchTerm) {
      performSearch(searchTerm, renderCallback);
    }
  });
```